plugins/inputs/snmp/snmp.go

```diff
@@ -149,6 +131,43 @@ func (s *Snmp) init() error {
 	return nil
 }
 
+func (s *Snmp) getMibsPath() error {
+	gosmi.Init()
+	var folders []string
+	for _, mibPath := range s.Path {
+		gosmi.AppendPath(mibPath)
+		folders = append(folders, mibPath)
+		err := filepath.Walk(mibPath, func(path string, info os.FileInfo, err error) error {
+			if info.Mode()&os.ModeSymlink != 0 {
+				s, _ := os.Readlink(path)
+				folders = append(folders, s)
```

Comment:
No error checking? Maybe only add the folder if it's OK?

Reply:
made it error and print out a warning